native/chat/chat-thread-list.react.js

```diff
@@ -124,6 +129,18 @@ class ChatThreadList extends React.PureComponent<Props, State> {
         />
       );
     }
+    if (item.type === 'empty') {
```

Comment:
Let's avoid having code here that is specific to background threads. Background thread stuff should live in `BackgroundChatThreadList`.

How about making `emptyItem` a `React.Node`, and passing this node in from `BackgroundChatThreadList`?

Reply:
To clarify: do you mean only the `Text` component with  explanation?

Reply:
Yeah, that's what I meant!

Let's use `item.emptyItem` here instead of `this.props.emptyItem`. In this case it actually makes no difference since they are the same, but as a rule of thumb `renderItem` shouldn't be looking at `this.props` because when `this.props` changes, there's no guarantee `renderItem` will be run again. `renderItem` only gets run again when the `item` objects change.

This is why we always try to put the props into the item. (In case where a single prop affects every item, an alternate approach is to pass that prop in as `extraData` to the `FlatList`.)

Reply:
I see one problem with using `item.emptyItem` here: when user changes theme, from light to dark for example, `listDataSelector` is not called again, `item.emptyItem` is not changing, and therefore until `listDataSelector` is called again, we're still displaying the old `item.emptyItem`.

Reply:
Great catch!! You're totally right, this is an issue here. The reason it's happening is because we're checking `this.props.emptyItem` directly in `listDataSelector`, instead of selecting it from `propsAndState`. I should've noticed this on the first review.

One concern about selecting it from `propsAndState` is that since we're generating a new `React.Node` in `ChatThreadBackground` every time it renders, that means we will end up generating a new `data` for the `FlatList` in `ChatThreadList` every time `ChatThreadBackground` renders.

Here's an idea - instead of passing in a `React.Node`, let's pass in a React component (`React.ComponentType`). We can define a component for our `emptyItem` that calls `useStyles`, and pass that component in from `ChatThreadBackground` to `ChatThreadList`. Then that component will handle re-rendering itself when dark mode changes, without us having to rerender the entire `FlatList`.